core/android_make.go

```diff
@@ -667,89 +662,92 @@ func (g *androidMkGenerator) generateCommonActions(m *generateCommon, ctx bluepr
-		text += outs + ": depfile := " + inout.depfile + "\n"
-		text += outs + ": " + ins + " " + strings.Join(inout.implicitSrcs, " ") + "\n"
-		text += "\t" + cmd + "\n"
+		sb.WriteString(outs + ": in := " + ins + "\n")
+		sb.WriteString(outs + ": out := " + strings.Join(inout.out, " ") + "\n")
+		sb.WriteString(outs + ": depfile := " + inout.depfile + "\n")
+		sb.WriteString(outs + ": " + ins + " " + strings.Join(inout.implicitSrcs, " ") + "\n")
+		sb.WriteString("\t" + cmd + "\n")
 		if inout.depfile != "" {
 			// Convert the depfile file format as part of this rule.
-			text += g.transformDepFile("$(depfile)")
+			sb.WriteString(g.transformDepFile("$(depfile)"))
 			// ...and include it outside the rule.
-			text += g.includeDepFile(outs, inout.depfile)
+			sb.WriteString(g.includeDepFile(outs, inout.depfile))
 		}
-		text += outputsVarName(m) + " += " + outs + "\n"
+		sb.WriteString(outputsVarName(m) + " += " + outs + "\n")
 
 		for _, out := range inout.out[1:] {
-			text += out + ": " + outs + "\n"
-			text += outputsVarName(m) + " += " + out + "\n"
+			sb.WriteString(out + ": " + outs + "\n")
+			sb.WriteString(outputsVarName(m) + " += " + out + "\n")
 		}
-		text += "\n"
+		sb.WriteString("\n")
 	}
-	text += g.outputs(m) + ": " + strings.Join(implicits, " ") + "\n"
+	sb.WriteString(g.outputs(m) + ": " + strings.Join(implicits, " ") + "\n")
 
 	/* This will ensure that any dependencies will not be rebuilt in the case of no change */
-	text += ".KATI_RESTAT: " + g.outputs(m) + "\n"
-
-	return text
+	sb.WriteString(".KATI_RESTAT: " + g.outputs(m) + "\n")
 }
 
 func (g *androidMkGenerator) generateSourceActions(m *generateSource, ctx blueprint.ModuleContext, inouts []inout) {
 	if enabledAndRequired(m) {
-		text := g.generateCommonActions(&m.generateCommon, ctx, inouts)
-		text += installGeneratedFiles(m, ctx, m.generateCommon.Properties.Tags)
-		androidMkWriteString(ctx, m.altShortName(), text)
+		var sb strings.Builder
+		g.generateCommonActions(&sb, &m.generateCommon, ctx, inouts)
+		installGeneratedFiles(&sb, m, ctx, m.generateCommon.Properties.Tags)
+		androidMkWriteString(&sb, ctx, m.altShortName())
 	}
 }
 
 func (g *androidMkGenerator) transformSourceActions(m *transformSource, ctx blueprint.ModuleContext, inouts []inout) {
 	if enabledAndRequired(m) {
-		text := g.generateCommonActions(&m.generateCommon, ctx, inouts)
-		text += installGeneratedFiles(m, ctx, m.generateCommon.Properties.Tags)
-		androidMkWriteString(ctx, m.altShortName(), text)
+		var sb strings.Builder
+		g.generateCommonActions(&sb, &m.generateCommon, ctx, inouts)
+		installGeneratedFiles(&sb, m, ctx, m.generateCommon.Properties.Tags)
+		androidMkWriteString(&sb, ctx, m.altShortName())
 	}
 }
 
 func (g *androidMkGenerator) genStaticActions(m *generateStaticLibrary, ctx blueprint.ModuleContext, inouts []inout) {
 	if enabledAndRequired(m) {
-		text := g.generateCommonActions(&m.generateCommon, ctx, inouts)
+		var sb strings.Builder
```

Comment:
I wonder if you could declare it as `sb := &strings.Builder{}`?

That way, it will always be a pointer, so you can pass it to the functions below as `sb` (instead of `&sb`).